app/controllers/api/v2/hosts_controller.rb

```
@@ -280,8 +280,12 @@ def boot
       param :facts, Hash,      :required => true, :desc => N_("hash containing the facts for the host")
       param :certname, String, :desc => N_("optional: certname of the host")
       param :type, String,     :desc => N_("optional: the STI type of host to create")
-
       def facts
+        unless Setting[:enable_facts_storage]
+          render_message _("Fact Values feature is disabled, you can enable it in Settings > General"), :status => :ok
```

Should also return an error